src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/SourceControl/VSSourceControlManagerProvider.cs

```diff
@@ -15,41 +15,39 @@ namespace NuGet.PackageManagement.VisualStudio
     internal sealed class VsSourceControlManagerProvider : ISourceControlManagerProvider
     {
         private const string TfsProviderName = "{4CA58AB2-18FA-4F8D-95D4-32DDF27D184C}";
-
+        private readonly IAsyncServiceProvider _asyncServiceProvider;
         private readonly Configuration.ISettings _settings;
-        private readonly Lazy<EnvDTE.DTE> _dte;
 
         [ImportingConstructor]
         public VsSourceControlManagerProvider(
             [Import(typeof(SVsServiceProvider))]
-            IServiceProvider serviceProvider,
+            IAsyncServiceProvider asyncServiceProvider,
             Configuration.ISettings vsSettings)
         {
-            Assumes.Present(serviceProvider);
+            Assumes.NotNull(asyncServiceProvider);
+            _asyncServiceProvider = asyncServiceProvider;
             Assumes.Present(vsSettings);
 
             _settings = vsSettings;
-
-            _dte = new Lazy<EnvDTE.DTE>(
```

Comment:
Was the removal of the DTE caching intentional?

Reply:
Sorry, not intentional. Replaced with AsyncLazy, please check if implementation is correct.